cmd/pj-rehearse/main_test.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/ghodss/yaml"
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/watch"
+
+	clienttesting "k8s.io/client-go/testing"
+
+	"k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/client/clientset/versioned/fake"
+	prowconfig "k8s.io/test-infra/prow/config"
+	prowgithub "k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/kube"
+	pjdwapi "k8s.io/test-infra/prow/pod-utils/downwardapi"
+
+	"github.com/openshift/ci-operator-prowgen/pkg/diffs"
+	"github.com/openshift/ci-operator-prowgen/pkg/rehearse"
+)
+
+const (
+	candidatePath    = "../../test/pj-rehearse-integration/candidate"
+	masterPath       = "../../test/pj-rehearse-integration/master"
+	expectedJobsFile = "../../test/pj-rehearse-integration/expected_jobs.yaml"
+)
+
+var (
+	candidateConfigPath    = filepath.Join(candidatePath, diffs.ConfigInRepoPath)
+	candidateJobConfigPath = filepath.Join(candidatePath, diffs.JobConfigInRepoPath)
+	masterConfigPath       = filepath.Join(masterPath, diffs.ConfigInRepoPath)
+	masterJobConfigPath    = filepath.Join(masterPath, diffs.JobConfigInRepoPath)
+	logger                 = logrus.WithFields(logrus.Fields{prowgithub.OrgLogField: jobSpec.Refs.Org, prowgithub.RepoLogField: jobSpec.Refs.Repo})
+	loggers                = rehearse.Loggers{Job: logger, Debug: logger}
+	jobSpec                = pjdwapi.JobSpec{
+		Type:      kube.PresubmitJob,
+		Job:       "pull-pj-rehearse-demo",
+		BuildID:   "0",
+		ProwJobID: "prowjob",
+		Refs: &kube.Refs{
+			Org:     "super",
+			Repo:    "duper",
+			BaseRef: "base-ref",
+			BaseSHA: "base-sha",
+			Pulls: []v1.Pull{{
+				Number: 1234,
+				Author: "dptp-team",
+				SHA:    "a4dc6278a189aa66ae9c2a0f11834499f104fcd4",
+			}},
+		},
+	}
+
+	fakecs     = fake.NewSimpleClientset()
+	fakeclient = fakecs.ProwV1().ProwJobs("test-namespace")
+)
+
+func TestIntergration(t *testing.T) {
+	changedPresubmits := getChangedPresubmitsHelper(t)
+
+	watcher, err := fakeclient.Watch(metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("Failed to setup watch: %v", err)
+	}
+
+	expectedJobs := getExpectedProwJobs(t)
+	fakecs.Fake.PrependWatchReactor("prowjobs", func(clienttesting.Action) (bool, watch.Interface, error) {
+		watcher.Stop()
+		ret := watch.NewFakeWithChanSize(expectedJobs.Len(), true)
+
+		for event := range watcher.ResultChan() {
+			pj := event.Object.(*v1.ProwJob).DeepCopy()
+
+			if !expectedJobs.Has(pj.Spec.Job) {
+				t.Fatalf("Job %s shouldn't be rehearsed", pj.Spec.Job)
```

Comment:
Do we test the obverse? That all jobs we want to see rehearsed, actually get rehearsed?

Reply:
addressed